test/model.test.js

```diff
@@ -7575,6 +7575,16 @@ describe('Model', function() {
 
     assert.strictEqual(doc.__v, 0);
   });
+  it('updates the model\'s db property to point to the provided connection instance and vice versa', async function() {
+    const schema = new mongoose.Schema({
+      name: String
+    });
+    const Model = db.model('Test', schema);
+    const connection = start();
```

Comment:
We should expand this test to use a different uri than `db`, and assert that `Model.find()` executes queries against the new uri rather than the old uri. 

As written, I don't think that will work because on further review we'll need to also update the model's collection properties (see `Model.compile()`)

```
  const collection = connection.collection(
    collectionName,
    collectionOptions
  );

  model.prototype.collection = collection;
  model.prototype.$collection = collection;
  model.prototype[modelCollectionSymbol] = collection;
```